src/lcm/grids.py

```diff
@@ -16,37 +16,56 @@ class Grid(ABC):
     """LCM Grid base class."""
 
     @abstractmethod
-    def to_jax(self) -> jnp.ndarray:
+    def to_jax(self) -> Array:
         """Convert the grid to a Jax array."""
 
 
-@dataclass(frozen=True)
 class DiscreteGrid(Grid):
-    """A grid of discrete values.
+    """A class representing a discrete grid.
+
+    Args:
+        category_class (type): The category class representing the grid categories. Must
+            be a dataclass with fields that have unique scalar int or float values.
 
     Attributes:
-        options: The options in the grid. Must be an iterable of scalar int or float
-            values.
+        categories: The list of category names.
+        codes: The list of category codes.
+
+    Raises:
+        GridInitializationError: If the `category_class` is not a dataclass with scalar
+            int or float fields.
 
     """
 
-    options: Collection[int | float]
+    def __init__(self, category_class: type) -> None:
```

Comment:
why `type`? I would have expected `dataclass`.

Reply:
Unfortunately `dataclass` is not a type. And as we input the class itself and not an instance of the class, we cannot assume more than `type`.

```python
@dataclass
class Health:
   pass

DiscreteGrid(Health)  # here Health is the class and an instance
```